### ISSUE TITLE
[package] boost/1.75.0: Header files not found after linking to Boost::headers target

### ISSUE THREAD

**Issue by tjwrona** [conan-io/conan-center-index#4064]
I am using the "boost/1.75.0" package in a project with the "cmake" and "cmake_paths" generators.

In the "CMakeLists.txt" file I find and link to the package with:

```cmake
find_package(Boost REQUIRED)

# ... create a target

target_link_libraries(<target> PRIVATE Boost::headers)
```

I am trying to use the <boost/type_index.hpp> header in a cpp file for this target, but the header file cannot be found. As far as I know the header paths should be part of the target so they should automatically be resolved correctly.

### Package and Environment Details (include every applicable attribute)
  * Package Name/Version: **boost/1.75.0**
  * Operating System+version: **Windows 10**
  * Compiler+version: **MSVC 16.9.0.57004**
  * Conan version: **conan 1.29.2**
  * Python version: **Python 3.7.1**

**Comment by tjwrona1992** [comment]
Note that I tried using the "cmake_find_package" generator, but that makes one of my other dependencies break so that isn't an option. All of my other dependencies work with the "cmake_paths" generator so I rely heavily on that and I would expect Boost to work with that generator as well.

**Comment by madebr** [comment]
Can you please check whether the file `boost/type_index.hpp` is in the boost package?

I want to know whether the problem is a wrong include path or missing header(s).

**Comment by tjwrona1992** [comment]
It is there, but in a very strange location:

`C:\.conan\e92f26\1\source_subfolder\boost\type_index.hpp`

The actual package contains a ".conan_link" file here:

`C:\Users\tjwro\.conan\data\boost\1.75.0\_\_\source`

Inside the ".conan_link" file it points to:

`C:\.conan\e92f26\1`

**Comment by madebr** [comment]
The `C:\.conan/e92f26\1` thing is due to [`short_paths`](https://docs.conan.io/en/latest/reference/conanfile/attributes.html#short-paths).
It's a conan feature to shorten the paths.
This can avoid a `PATH_MAX` error on Windows.

You checked the wrong folder. You need to check `C:\Users\tjwro\.conan\data\boost\1.75.0\_\_\package\<some hash>` -> `<path from .conan_link>`.

When building a package, conan uses 3 folders:
- a source folder to put the boost sources
- a build folder where to build boost
- a package folder where to install boost to (this is the folder that your application should use)

I just checked the `boost/1.75.0` conan package on my Linux box and `include/boost/type_index.hpp` is there.

**Comment by tjwrona1992** [comment]
Looking at the path you mentioned, the `.conan_link` file points to here:

`C:\.conan\cead48\1`

Which has the following in it:

`include\boost\type_index.hpp`

So it looks like the header is there.

I am using Visual Studio, but I'm using the latest version (2019 16.9.0 Preview 2.0) which supports directly opening CMake projects without using the "Visual Studio" CMake generator first. As far as I can tell there are no Visual Studio project or solution files being generated.

Upon further investigation though, I tried changing the `find_package` call to explicitly use the `CONFIG` option... `find_package(Boost CONFIG REQUIRED)` and that caused the call to `find_package` to fail. It looks like no package config files are installed with the Boost conan package, so the "cmake_paths" generator will not work properly with it.

**Comment by madebr** [comment]
I didn't read your first sentence.
You need to set as generators: `cmake` and `cmake_find_package`.
The first will generate `conanbuildinfo.cmake`, the latter `FindBoost.cmake`.

**conanfile.txt**
```ini
[requires]
boost/1.75.0
[generators]
cmake
cmake_find_package
```
**CMakeLists.txt**
```cmake
cmake_minimum_required(VERSION 3.0)
project(boost_header_test)
list(APPEND CMAKE_MODULE_PATH "${CMAKE_BINARY_DIR}")
include(conanbuildinfo)
conan_basic_setup()
find_package(Boost REQUIRED)
add_executable(main main.cpp)
target_link_libraries(main PRIVATE Boost::headers)
```
**main.cpp**
```c++
#include <boost/type_index.hpp>
int main() {}
```

Copy to some folder and run:
```sh
mkdir build
cd build
conan install ..
cmake ..
cmake --build . --config Release
```

**Comment by tjwrona1992** [comment]
Okay I got it to work, thanks for the help!

I initially had some issues once I added the `cmake_find_package` generator because it broke all of my packages that used the `cmake_paths` generator.

To fix it, I had to specify `CONFIG` in all of the `find_package` calls for the packages that are being consumed with the `cmake_paths` generator (`find_package(<package> CONFIG REQUIRED)`). Once I did that I was able to use both generators in the same project and things work okay.
